zonemaster/zonemaster-engine
# Update Makefile solution in share to make it compatible with FreeBSD

## Purpose

FreeBSD does not use GNU Make file, and it is too complicated to create a Makefile compatible with both. This PR adds a wrapper Makefile in share and renames the real Makefile to GNUmakefile, which is the default GNU Make Makefile file name. Now both Linux and FreeBSD users will be happy!

## Context

Fixes #702 for real

## Changes

* Rename Makefile to GNUmakefile and add Makefile as wrapper for FreeBSD.
* Update .gitignore so that it does not ignore Makefile in `share` directory plus some editorial updates there.
* Updates MANIFEST with new file
* Updates MANIFEST.SKIP with .github directory (unrelated to the purpose of this PR, relates to PR #890 that forgot to add it)
* Updates Makefile.PL to permit choice of gmake for FreeBSD via environment variable (standard procedure)
* Removes special commands for FreeBSD in translation document
* Removes special FreeBSD setting in  t/po-files.t 

## How to test this PR

* Verify that `perl Makefile;make all;make distcheck` does not complain on missing MO files (FreeBSD or Linux)
* Verify that operations in `share` directory, such as `make extract-pot`, `make sv.po` and `make check-msg-args` and works under both FreeBSD and Linux.
